packages/helpers/classes/personalization.js

```diff
@@ -275,43 +275,26 @@ class Personalization {
    * To JSON
    */
   toJSON() {
-
     //Get data from self
-    const {
-      to, cc, bcc, subject, headers, customArgs, sendAt,
-      substitutions, substitutionWrappers,
-    } = this;
+    const {to, cc, bcc, subject, headers, customArgs, sendAt, substitutions, substitutionWrappers} = this;
 
     //Initialize with mandatory values
     const json = {to};
 
     //Arrays
-    if (Array.isArray(cc) && cc.length > 0) {
-      json.cc = cc;
-    }
-    if (Array.isArray(bcc) && bcc.length > 0) {
-      json.bcc = bcc;
-    }
-
+    json.cc = Array.isArray(cc) && cc.length > 0 ? cc : null;
+    json.bcc = Array.isArray(bcc) && bcc.length > 0 ? bcc : null;
     //Objects
-    if (Object.keys(headers).length > 0) {
-      json.headers = headers;
-    }
+    json.headers = Object.keys(headers).length > 0 ? headers : null;
+    json.customArgs = Object.keys(customArgs).length > 0 ? customArgs : null;
+    //Simple properties
+    json.subject = typeof subject !== 'undefined' ? subject : null;
+    json.sendAt = typeof sendAt !== 'undefined' ? sendAt : null;
```

Comment:
Also you neednt put a typeof check for undefined..
```javascript
json.sendAt = sendAt ? sendAt : null;
```

Reply:
The problem is that there should not be a property set on the `json` object at all if there is no valid value, e.g. no null, no undefined. In your above example `json.subject` will be `null` and that will break the API.

Reply:
Yes agreed so it can be modified to 
```javascript
json.sendAt = sendAt ? sendAt : '';
```

Reply:
Nope :) That will have the same effect. The key will still be set.